clusterloader2/pkg/measurement/common/network/network_performance_measurement.go

```diff
@@ -0,0 +1,397 @@
+}
+
+func (npm *networkPerformanceMeasurement) createAndWaitForWorkerPods() error {
+	// Create worker pods
+	var replicas = map[string]interface{}{"Replicas": npm.numberOfClients + npm.numberOfServers}
+	if err := npm.framework.ApplyTemplatedManifests(workerPodDeploymentManifestFilePath, replicas); err != nil {
+		return fmt.Errorf("error while creating worker pods :%v ", err)
+	}
+	// Wait for all worker pods to be ready
+	stopCh := make(chan struct{})
+	time.AfterFunc(podReadyTimeout, func() {
+		close(stopCh)
+	})
+	options := &measurementutil.WaitForPodOptions{
+		Selector:            &measurementutil.ObjectSelector{Namespace: netperfNamespace},
+		DesiredPodCount:     npm.numberOfClients + npm.numberOfServers,
+		CallerName:          networkPerformanceMetricsName,
+		WaitForPodsInterval: 2 * time.Second,
+	}
+	return measurementutil.WaitForPods(npm.k8sClient, stopCh, options)
+}
+
+func (*networkPerformanceMeasurement) String() string {
+	return networkPerformanceMetricsName
+}
+
+func newWorkerPodData(podName, nodeName, podIP string) *workerPodData {
+	return &workerPodData{podName: podName, workerNode: nodeName, podIP: podIP}
+}
+
+func (npm *networkPerformanceMeasurement) storeWorkerPods() error {
+	pods, err := npm.k8sClient.CoreV1().Pods(netperfNamespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for _, pod := range pods.Items {
+		if pod.Status.PodIP == "" {
+			klog.Errorf("IP address not found for %s", pod.Name)
+			continue
+		}
+		npm.addPodWorker(newWorkerPodData(pod.Name, pod.Spec.NodeName, pod.Status.PodIP))
+	}
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) gather() (measurement.Summary, error) {
+	close(npm.stopCh)
+	npm.cleanupCluster()
+	resultSummary := npm.createResultSummary()
+	content, err := util.PrettyPrintJSON(&measurementutil.PerfData{
+		Version:   "v1",
+		DataItems: resultSummary.DataItems,
+	})
+	if err != nil {
+		klog.Info("Pretty Print to Json Err: ", err)
+	}
+	summaryName := fmt.Sprintf(npm.String() + resultSummary.podRatio + resultSummary.protocol + resultSummary.service)
+	return measurement.CreateSummary(summaryName, "json", content), nil
+}
+
+// Dispose disposes resources,objects after the measurement.
+func (npm *networkPerformanceMeasurement) Dispose() {
+	if npm.framework == nil {
+		klog.V(1).Infof("Network measurement %s wasn't started, skipping the Dispose() step", npm)
+		return
+	}
+	klog.Infof("Stopping network performance measurement...")
+	npm.cleanupCluster()
+}
+
+func (npm *networkPerformanceMeasurement) handleUpdateNetworkTestEvents(newObj interface{}) {
+	newRuntimeObj, ok := newObj.(runtime.Object)
+	if newObj != nil && !ok {
+		klog.Errorf("error casting object: %s", newObj)
+		return
+	}
+	// TODO(#1757): Stop relying on unstructured data.
+	resourceContent, err := runtime.DefaultUnstructuredConverter.ToUnstructured(newRuntimeObj)
+	if err != nil {
+		klog.Errorf("error converting event Object to map representation: %s", newObj)
```

Comment:
Failed to convert to unstructured: %v", err

Reply:
modified